### PR TITLE
Transactions List on Investment Webpage

### PR DESCRIPTION
**Pull request by cindyzhang12**
[//]: # "These comments meant for your reference, they are invisible and don't need to be deleted"

## What's new in this PR?
- implemented a list of all buy shares and dividends transactions on my investment page!

### How to Review
- review TransactionsList.js for component, Investment.css for styling, and Investments.js for implementation

### Related PRs
N/A

### Next steps
- make sure it works with all edge cases which includes filtering through bill payments (this should work though), and positive values for dividend transactions, and anything else!

### Screenshots

[//]: # '#####################################'
[//]: # '#### YOUR SCREENSHOTS BELOW ########'
[//]: # '############## END ##################'

### Design Status

[//]: # 'If this is a frontend PR, what is the expected 
status of the UI in this PR (lo, mid, high- fi?'
[//]: # '#####################################'
[//]: # '####### LOFI/MIDFI/HIFI ###########'
[//]: # '############## END ##################'

CC: @dfangshuo

### CODE REVIEW

File: src/styles/Investments.css [hunk 1]
```diff
@@ -214,3 +214,61 @@
...
+}
+
+.transaction-card-details p {
+  font-size: 2.3vh;
+  line-height: 22px;
+  font-weight: normal;
+  margin-block-start: 0.7em;
+  color: var(--pp-black);
+  letter-spacing: 0.01em;
+}
```
Comment (dfangshuo) [inline src/styles/Investments.css:265]: make these names more specific, pre-append investments-page etc. for instance

File: src/screens/general/Investment.js [hunk 2]
```diff
@@ -107,11 +117,13 @@ class Investment extends React.PureComponent {
...
               <h2>Transactions</h2>
-              <div className="transactions-box">transactions box test</div>
+              <div className="transactions-box">
+                <TransactionList payments={payments} />
+              </div>
             </div>
             <div className="right-content">
               <h2>Financial Breakdown</h2>
-              <div className="fin-box">fin box test</div>
+              <div className="fin-box">adf</div>
```
Comment (dfangshuo) [inline src/screens/general/Investment.js:126]: what is the adf?

File: src/screens/general/Investment.js [hunk 3]
```diff
@@ -47,9 +50,16 @@ class Investment extends React.PureComponent {
     await refreshUserData(owner.id);
   };
 
+  getPayments = async () => {
+    const { owner } = this.props;
+    let paymentsList = [];
+    paymentsList = await getPaymentsByIds(owner.paymentIds || []);
```
Comment (niwong) [inline src/screens/general/Investment.js:56]: How is this comparing `owner.paymentIds` and the empty array? Are you saying that if `owner.paymentIds` is empty return an empty array?

File: src/screens/general/components/TransactionsList.js [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+          <div className="investments-transaction-card-details">
+            {type === 'Dividend' ? null : <p>Quantity: {quantity}</p>}
+            <h6>{dateFormatted}</h6>
+            <hr />
+          </div>
+        </div>
+      );
+    });
+
+    return <div style={{ overflow: 'auto' }}>{list}</div>;
```
Comment (niwong) [inline src/screens/general/components/TransactionsList.js:32]: Looks great! I'd suggest adding this inline style into a class and corresponding CSS file! 👍🏼 Nice job!

File: src/styles/Investments.css [hunk 5]
```diff
@@ -214,3 +214,61 @@
...
+.investments-transaction-card {
+  background: white;
+  border-radius: 5px;
+  padding: 35px;
+  margin: 10px 0px;
+  display: flex;
+  justify-content: space-between;
+  flex-direction: column;
+  padding-bottom: 0px;
+  padding-top: 0px;
```
Comment (niwong) [inline src/styles/Investments.css:227]: If you want the left and right padding to be 35px and the top and bottom padding to be 0px, you can use the padding CSS "shorthand" and write it like this: `padding: 0px 25px`. This specifies `padding: [top & bot] [left & right]

File: src/styles/Investments.css [hunk 6]
```diff
@@ -214,3 +214,61 @@
   display: flex;
   justify-content: flex-end;
 }
+
+.investments-transaction-card {
+  background: white;
```
Comment (niwong) [inline src/styles/Investments.css:219]: Use color variable

**Comment by niwong** [comment]
Nice job Cindy! Just some small stuff, but overall looks amazing.

File: src/screens/general/Investment.js (hunk 3 above)
Comment (cindyzhang12) [reply src/screens/general/Investment.js:59]: yeah! if someone doesn't have any paymentIds, then owner.paymentIds returns null and it errors, so if it's null it should be an empty array instead! :)

File: src/styles/Investments.css (hunk 6 above)
Comment (cindyzhang12) [reply src/styles/Investments.css:223]: do you mean var(--pp-white)? for some reason when i use this it turns out to be a weird gray :'))

Comment (dfangshuo) [reply src/styles/Investments.css:223]: white is fine as well haha

**Pull request merged**